torchci/scripts/check_alerts.py

```diff
@@ -306,32 +306,41 @@ def main():
     (jobs_to_alert_on, flaky_jobs) = classify_jobs(job_names, sha_grid)
 
     # Fetch alerts
-    alerts = fetch_alerts()
-    alerts_cleared = False
+    existing_alerts = fetch_alerts()
 
-    # Alerts should be singletons and there should only be 1 alert
     # Alerts should also be cleared if the current status of HUD is green
-    if len(alerts) > 1 or should_clear_alerts(sha_grid):
-        alerts_cleared = clear_alerts(alerts)
+    if len(jobs_to_alert_on) == 0 and should_clear_alerts(sha_grid):
+        if clear_alerts(existing_alerts):
+            existing_alerts = [] # all alerts have now been cleared
 
-    # Create a new alert if no alerts active or edit the original one if there's a new update
-    no_alert_currently_active = alerts_cleared or len(alerts) == 0
-    if len(jobs_to_alert_on) > 0 and (no_alert_currently_active):
-        create_issue(generate_failed_job_issue(jobs_to_alert_on))
-    elif len(jobs_to_alert_on) > 0 and len(alerts) == 1:
-        new_issue = generate_failed_job_issue(jobs_to_alert_on)
-        if alerts[0]["body"] != new_issue["body"]:
-            update_issue(new_issue, alerts[0]["number"])
-        else:
-            print("No new updates. Not updating any alerts.")
-    else:
+    if len(jobs_to_alert_on) == 0:
         print(
             "Didn't find anything to alert on.",
             no_alert_currently_active,
             jobs_to_alert_on,
         )
-    # TODO: Record flaky jobs in rockset or something re run or analyze
+        return
+
+    # Find the existing alert for recurrently failing jobs (if any).
+    # We're going to update the existing alert if possible instead of filing a new one.
+    existing_recurrent_job_failure_alert = None
```

Comment:
Nit: one liner if you prefer `existing_recurrent_job_failure_alert = next((alert for alert in existing_alerts if "Recurrently..." in alert["title"]), None)`

Reply:
Pretty cool :) 

The current mode might be more verbose, but it requires less python-fu for me to grok 😅

I think I'll leave this one as is